app/integration_action.go

```diff
@@ -207,6 +207,23 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 // Perform an HTTP POST request to an integration's action endpoint.
 // Caller must consume and close returned http.Response as necessary.
 func (a *App) DoActionRequest(rawURL string, body []byte) (*http.Response, *model.AppError) {
+	inURL, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, model.NewAppError("DoActionRequest", "api.post.do_action.action_integration.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+
+	siteURL, _ := url.Parse(*a.Config().ServiceSettings.SiteURL)
+	rawURLPath := path.Clean(rawURL)
```

Comment:
Wondering if we should only start parsing `rawURL` as a pure path if `inURL.Host == ""`? That would also make it a little more obvious in the code that we're dealing with a pure path.

Reply:
The problem I found is that ```url.Parse()``` doesn't work well with "dirty" paths and will fail in such cases. For example, if you try to parse the relative path ```//plugins/myplugin/myaction``` you are going to get a URL object with ```Host``` set to ```plugins```. That's why I chose to clean it from the raw string and not checking for the parsing result.

Reply:
Ah, hmm, that seems like it's going to break in the inverse as well:
```go
package main

import (
	"fmt"
	"path"
)

func main() {
	fmt.Println(path.Clean("http://google.ca"))
}
```

outputs:
```
http:/google.ca
```

Perhaps we can simplify all of this to just require the path to start with a leading `/`, and if it doesn't, let it be parsed as a URL.

Reply:
Yes it does but it's not going to lead to a failure cause that cleaned path is only used if it starts with either "plugins" or "/plugins".
That said I agree this is getting a bit contrived.

Reply:
Ah, you're right re: only being used then. All good!